### ISSUE TITLE
💅 Import used by decorator metadata reformatted as type import since Biome 1.6.0

### ISSUE THREAD

**Issue by bc-m** [biomejs/biome#2038]
### Environment information

```bash
CLI:
  Version:                      1.6.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        complexity/noForEach = "warn"
                                complexity/noUselessSwitchCase = "off"
                                correctness/noUnnecessaryContinue = "off"
                                style/noInferrableTypes = "off"
                                style/noNonNullAssertion = "warn"
                                style/useNodejsImportProtocol = "error"
                                suspicious/noExplicitAny = "warn"
```

### Rule name

lint/style/useImportType

### Playground link

https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAQgBvAGQAeQAsACAAQwBvAG4AdAByAG8AbABsAGUAcgAsACAARwBlAHQALAAgAFAAYQByAGEAbQAsACAAUABvAHMAdAAsACAAVQBzAGUARwB1AGEAcgBkAHMAIAB9ACAAZgByAG8AbQAgACcAQABuAGUAcwB0AGoAcwAvAGMAbwBtAG0AbwBuACcAOwAKAGkAbQBwAG8AcgB0ACAAewAgAFIAbwBsAGUAcwAgAH0AIABmAHIAbwBtACAAJwAuAC4ALwBjAG8AbQBtAG8AbgAvAGQAZQBjAG8AcgBhAHQAbwByAHMALwByAG8AbABlAHMALgBkAGUAYwBvAHIAYQB0AG8AcgAnADsACgBpAG0AcABvAHIAdAAgAHsAIABSAG8AbABlAHMARwB1AGEAcgBkACAAfQAgAGYAcgBvAG0AIAAnAC4ALgAvAGMAbwBtAG0AbwBuAC8AZwB1AGEAcgBkAHMALwByAG8AbABlAHMALgBnAHUAYQByAGQAJwA7AAoAaQBtAHAAbwByAHQAIAB7ACAAUABhAHIAcwBlAEkAbgB0AFAAaQBwAGUAIAB9ACAAZgByAG8AbQAgACcALgAuAC8AYwBvAG0AbQBvAG4ALwBwAGkAcABlAHMALwBwAGEAcgBzAGUALQBpAG4AdAAuAHAAaQBwAGUAJwA7AAoAaQBtAHAAbwByAHQAIAB7ACAAQwBhAHQAcwBTAGUAcgB2AGkAYwBlACAAfQAgAGYAcgBvAG0AIAAnAC4ALwBjAGEAdABzAC4AcwBlAHIAdgBpAGMAZQAnADsACgBpAG0AcABvAHIAdAAgAHQAeQBwAGUAIAB7ACAAQwByAGUAYQB0AGUAQwBhAHQARAB0AG8AIAB9ACAAZgByAG8AbQAgACcALgAvAGQAdABvAC8AYwByAGUAYQB0AGUALQBjAGEAdAAuAGQAdABvACcAOwAKAGkAbQBwAG8AcgB0ACAAdAB5AHAAZQAgAHsAIABDAGEAdAAgAH0AIABmAHIAbwBtACAAJwAuAC8AaQBuAHQAZQByAGYAYQBjAGUAcwAvAGMAYQB0AC4AaQBuAHQAZQByAGYAYQBjAGUAJwA7AAoACgBAAFUAcwBlAEcAdQBhAHIAZABzACgAUgBvAGwAZQBzAEcAdQBhAHIAZAApAAoAQABDAG8AbgB0AHIAbwBsAGwAZQByACgAJwBjAGEAdABzACcAKQAKAGUAeABwAG8AcgB0ACAAYwBsAGEAcwBzACAAQwBhAHQAcwBDAG8AbgB0AHIAbwBsAGwAZQByACAAewAKACAAIABjAG8AbgBzAHQAcgB1AGMAdABvAHIAKABwAHIAaQB2AGEAdABlACAAcgBlAGEAZABvAG4AbAB5ACAAYwBhAHQAcwBTAGUAcgB2AGkAYwBlADoAIABDAGEAdABzAFMAZQByAHYAaQBjAGUAKQAgAHsAfQAKAAoAIAAgAEAAUABvAHMAdAAoACkACgAgACAAQABSAG8AbABlAHMAKABbACcAYQBkAG0AaQBuACcAXQApAAoAIAAgAGMAcgBlAGEAdABlACgAQABCAG8AZAB5ACgAKQAgAGMAcgBlAGEAdABlAEMAYQB0AEQAdABvADoAIABDAHIAZQBhAHQAZQBDAGEAdABEAHQAbwApACAAewAKACAAIAAgACAAdABoAGkAcwAuAGMAYQB0AHMAUwBlAHIAdgBpAGMAZQAuAGMAcgBlAGEAdABlACgAYwByAGUAYQB0AGUAQwBhAHQARAB0AG8AKQA7AAoAIAAgAH0ACgAKACAAIABAAEcAZQB0ACgAKQAKACAAIABmAGkAbgBkAEEAbABsACgAKQA6ACAAUAByAG8AbQBpAHMAZQA8AEMAYQB0AFsAXQA%2BACAAewAKACAAIAAgACAAcgBlAHQAdQByAG4AIAB0AGgAaQBzAC4AYwBhAHQAcwBTAGUAcgB2AGkAYwBlAC4AZgBpAG4AZABBAGwAbAAoACkAOwAKACAAIAB9AAoACgAgACAAQABHAGUAdAAoACcAOgBpAGQAJwApAAoAIAAgAGYAaQBuAGQATwBuAGUAKAAKACAAIAAgACAAQABQAGEAcgBhAG0AKAAnAGkAZAAnACwAIABuAGUAdwAgAFAAYQByAHMAZQBJAG4AdABQAGkAcABlACgAKQApAAoAIAAgACAAIABpAGQAOgAgAG4AdQBtAGIAZQByACwACgAgACAAKQAgAHsACgAgACAAIAAgAC8ALwAgAGcAZQB0ACAAYgB5ACAASQBEACAAbABvAGcAaQBjAAoAIAAgAH0ACgB9AA%3D%3D

### Expected result

Be satisfied with the file as it is, like in Biome v1.5.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Biome is not able to analyze decorator implementations. I don't see how we could detect that `CatsService` is actually being used. Have you some suggestions?

**Comment by bc-m** [comment]
> Biome is not able to analyze decorator implementations. I don't see how we could detect that `CatsService` is actually being used. Have you some suggestions?
>
> EDIT: We could ignore types used as a type annotation of a property parameter in a decorated class.

For a quick solution, I'm probably with your idea. I don't know the side effects yet.

For long term: Whats the state of the announced [type system](https://biomejs.dev/blog/roadmap-2024/#type-system)? Perhaps such a case can be covered by the type system in future?

**Comment by Conaclos** [comment]
> For a quick solution, I'm probably with your idea. I don't know the side effects yet.

We could miss some cases where a type qualifier is actually needed according to the rule.
However, using both property parameters and decorator seems a common pattern for dependency injection.
Thus, I think it is a fair change.

> For long term: Whats the state of the announced [type system](https://biomejs.dev/blog/roadmap-2024/#type-system)?

We have not started yet.

> Perhaps such a case can be covered by the type system in future?

I am unsure if we will handle such complex analysis.

**Comment by bc-m** [comment]
TIL that `@typescript-eslint/consistent-type-imports' does not work on NestJS files either, but it leaves the import as it is without displaying an error and without breaking the code after applying auto-fixes.

As a workaround, I disabled the `useImportType' rule in `biome.json'.

**Comment by vbasky** [comment]
@bc-m  I couldn't get the `useImportType` to work what is the version of the schema that supports that rule ?

**Comment by ematipico** [comment]
We recommend disabling the rule in angular/ nestjs projects: https://biomejs.dev/linter/rules/use-import-type/#caveat-with-typescript-experimental-decorators

**Comment by ruscon** [comment]
It would be nice to somehow solve this problem, because decorators are an important part of Typescript.

**Comment by kidonng** [comment]
[How (typescript-)eslint does it](https://typescript-eslint.io/rules/consistent-type-imports/#caveat-decorators--experimentaldecorators-true--emitdecoratormetadata-true), they simply skip this rule for files with decorators + `experimentalDecorators: true` + `emitDecoratorMetadata: true`. Biome could also do something alike.

**Comment by jpike88** [comment]
this one's real annoying, as I was simplying my biome.json file to align with recommended values and applying auto fixes, it screwed a lot of my angular files up meaning I have to start again.

And as I can't comment in a biome.json file to tell other why that rule is set to off.

Why not extend it so I can just whitelist a given import path?

**Comment by jpike88** [comment]
Can this issue be reopened please

**Comment by ematipico** [comment]
Feel free to open a new discussion to propose possible solutions
